.github/workflows/ci.yml

```diff
@@ -196,6 +196,13 @@ jobs:
             Remove-Item -Force -Recurse $kits\$kind\$sdk_version -ErrorAction Continue
           }
 
+      # Pre-build citool before the following step uninstalls rustup
+      # Build is into the build directory, to avoid modifying sources
+      - name: build citool
+        run: |
+          cd src/ci/citool
+          CARGO_TARGET_DIR=../../../build/citool cargo build
```

Comment:
on line 61 we run cargo test. It would be cool to share the compilation artifacts somehow.

Reply:
I'm not sure if it's worth the complexity.. I want to run the test before we kick off CI, to avoid a situation where we return something weird from the calculate matrix command and then CI would do something weird. The rebuild should take ~10s, I think that it would be genuinely slower to download, and would require additional CI steps here.